src/bgw/launcher_interface.c

```diff
@@ -29,3 +30,22 @@ bgw_num_unreserved(void)
 
 	return DatumGetInt32(DirectFunctionCall1(unreserved, BoolGetDatum(false))); /* no function call zero */
 }
+
+extern int
+bgw_loader_api_version(void)
+{
+	void	  **versionptr = find_rendezvous_variable(RENDEZVOUS_BGW_LOADER_API_VERSION);
```

Comment:
Yeah, so the reason I suggested a rendezvous variable was to avoid exposing the internal API version to the outside world. I guess now you are exposing it with a function anyway to be able to do the check in SQL, which kind of defeats my initial purpose of keeping this internal.

Maybe not a big deal, but is there a reason not to do the version check internally in the lib insted of handling it via a mix of SQL and C? For instance, you could do it in the `_PG_init()` function. We already do version checks for the extension and PG there so it would make sense to keep all such checks in one place.

Reply:
Yeah you are right, fixed.